brave/src/test/java/brave/features/propagation/SecondaryConditionalSamplingTest.java

```diff
@@ -0,0 +1,345 @@
+import brave.handler.FinishedSpanHandler;
+import brave.handler.MutableSpan;
+import brave.propagation.ExtraFieldPropagation;
+import brave.propagation.TraceContext;
+import brave.propagation.TraceContextOrSamplingFlags;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.After;
+import org.junit.Test;
+import zipkin2.reporter.Reporter;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.singletonList;
+import static java.util.Collections.singletonMap;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * This proves a concept needed for sites who have different tracing systems behind one Zipkin
+ * compatible endpoint. At a high level, normal probabilistic B3 sampling co-exists with Edge
+ * sampling (100% between the gateway and the first service behind it), and Triage sampling (100%
+ * conditional on request properties like a user ID). Our goal to allow this coexistence without
+ * redundant overhead or reporting queues and without sending irrelevant data to any of the
+ * systems.
+ *
+ * <h2>Zipkin vs Edge vs Triage</h2>
+ * In this example, there are three tracing systems: zipkin, edge and triage. The following points
+ * describe elaborate both what they need and what they don't need.
+ *
+ * <h3>Zipkin</h3>
+ * Zipkin is always on for the entire network, and if tracing occurs, it wants the entire trace from
+ * the first request to the last on every node in the network. Zipkin is the primary owner of the
+ * trace, and its state is in B3 headers, including the up-front decision on whether or not to
+ * record data on this request. To review, [B3 propagates](https://github.com/openzipkin/b3-propagation)
+ * through the process and across nodes, and the a sampling decision is trace-scoped: it never
+ * changes from true to false or visa versa.c
+ *
+ * <h3>Edge</h3>
+ * Edge tracing is traffic that starts at gateway of the architecture. Unlike Zipkin sampling, this
+ * is 100% with the only condition being TTL, a simple TTL of one hop-down. 100% trace data only
+ * including ingress and egress from the edge router can serve a lot of value as the teams can build
+ * reliable statistics and flow charts. It is easy to implement as it is relatively stateless. For
+ * example, the router always collects and the first node needs only to consume the instruction and
+ * expire the sampling decision given to them. (not propagate it further).
+ *
+ * <h3>Triage</h3>
+ * The triage system serves customer support goals. Like Edge, this is 100% data, but unlike Edge,
+ * where data is collected is conditional on request properties such as the customer ID and the
+ * services under investigation. In other words, the triage system is on-demand, and could cover a
+ * small subset of the architecture starting at an arbitrary place. As such, triage is more advanced
+ * that the other two systems.
+ *
+ * <h2>Trace data forwarder</h2>
```

Comment:
@cwensel @drolando @worldtiki @openzipkin/core  this example will introduce a familiar concept to you, trace data forwarder. In this case, it is different than "firehose handler" as the app is not responsible for deciding to send to two places. Instead, we push that responsibility into the zipkin-compatible endpoint at the cost of adding a tag to help that endpoint know where to divert data to (in some cases sending the same json to multiple places)